### PR TITLE
ddl: fix cancel truncate table ddl error

### PR DESCRIPTION
**Pull request by ciscoxll**
### What problem does this PR solve? 
We can only cancel truncate table job when the job state is on `JobStateNone` , if the job status is `JobStateDone` , the job will not be canceled.

### What is changed and how it works?
* truncate table is done in a transaction, if the job is not completed, it can be canceled.
* added tests.

### Check List 

Tests 

 - Unit test

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8504)

### CODE REVIEW

File: ddl/table.go [hunk 1]
```diff
@@ -264,6 +258,20 @@ func onTruncateTable(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ erro
 	return ver, nil
 }
 
+func checkTruncateTable(t *meta.Meta, job *model.Job) (*model.TableInfo, int64, error) {
```
Comment (shenli) [inline ddl/table.go]: Why call it `check`?

Comment (ciscoxll) [reply ddl/table.go]: This is a repetitive logic. It is also used in `rollingbackTruncateTable`.

Comment (ciscoxll) [reply ddl/table.go]: Can change the function name.

File: ddl/rollingback.go [hunk 2]
```diff
@@ -190,6 +190,21 @@ func rollingbackDropSchema(t *meta.Meta, job *model.Job) error {
...
 }
 
+func rollingbackTruncateTable(t *meta.Meta, job *model.Job) (ver int64, err error) {
+	tblInfo, _, err := checkTruncateTableExist(t, job)
+	if err != nil {
+		return ver, errors.Trace(err)
+	}
+
+	// Here truncate table is done in a transaction, if the job is not completed, it can be canceled.
+	if tblInfo.State == model.StatePublic {
```
Comment (winkyao) [inline ddl/rollingback.go]: you should judge the job.SchemaState, Truncate table will never change the tblInfo.State.

**Comment by ciscoxll** [comment]
@crazycs520 PTAL.

File: ddl/rollingback.go [hunk 3]
```diff
@@ -220,6 +220,20 @@ func rollingbackRebaseAutoID(t *meta.Meta, job *model.Job) (ver int64, err error
 	return cancelOnlyNotHandledJob(job)
 }
 
+func rollingbackTruncateTable(t *meta.Meta, job *model.Job) error {
+	_, err := getTableInfoAndCancelFaultJob(t, job, job.SchemaID)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	// If the job is not started, tidb can cancel this job.
+	if job.SchemaState == model.StateNone {
```
Comment (crazycs520) [inline ddl/rollingback.go]: use `cancelOnlyNotHandledJob`.

Comment (ciscoxll) [reply ddl/rollingback.go]: @crazycs520 Done.

**Comment by ciscoxll** [comment]
@crazycs520 PTAL.

**Comment by ciscoxll** [comment]
@crazycs520 PTAL.

**Comment by ciscoxll** [comment]
@crazycs520 PTAL.

**Comment by ciscoxll** [comment]
@crazycs520 PTAL.

**Pull request merged**